python/ray/data/dataset.py

```diff
@@ -1919,29 +1924,55 @@ def to_torch(self,
         instead of passing it into a torch ``DataLoader``.
 
         Each element in IterableDataset will be a tuple consisting of 2
-        elements. The first item is the features tensor, and the second item
-        is the label tensor. The features tensor will be of shape (N, n),
-        and the label tensor will be of shape (N, 1), where N is the
-        ``batch_size`` used by the DataLoader, and n is the number of features.
+        elements. The first item are the features, and the second item
+        is the label. Those can take on different forms, dependning on
+        the specified arguments.
+
+        For the features tensor (N is the ``batch_size`` used by the
+        DataLoader,  and n, m, k are the number of features per tensor):
+
+        * If ``feature_columns`` is a ``List[str]``, the features will be
+          a tensor of shape (N, n), with columns corresponding to =
+          ``feature_columns``
+
+        * If ``feature_columns`` is a ``List[List[str]]``, the features will be
+          a list of tensors of shape [(N, m),...,(N, k)], with columns of each
+          tensor corresponding to the elements of ``feature_columns``
+
+        * If ``feature_columns`` is a ``Dict[str, List[str]]``, the features
+          will be a dict of key-tensor pairs of shape
+          {key1: (N, m),..., keyN: (N, k)}, with columns of each
+          tensor corresponding to the value of ``feature_columns`` under the
+          key.
+
+        If ``unsqueeze_label_tensor=True`` (default), the label tensor will be
+        of shape (N, 1). Otherwise, it will be of shape (N,). It can also be
+        specifed as None for prediction (note that it will be still present
+        in the tuple yielded by the generator).
```

Comment:
> It can also be
        specifed as None for prediction (note that it will be still present
        in the tuple yielded by the generator).

Wording of this is a little weird - I thought this was referring to `unsqueeze_label_tensor`, not `label_column`.